src/functional_data.Rmd

```diff
@@ -0,0 +1,177 @@
+# Functional Data
+
+Functional data provides information about curves varying over a continuum, such as time.
+There are two commonly used approaches for analysing functional data.
+
+* Directly analyse the functional data using a [learner](&Learner.md) that is suitable for functional data on a [FDATask](&makeFDATask). Those learners have the prefixes __FDAClassif.__ and __FDARegr.__.
+
+* Transform the task into a format suitable for standard __classification__ or __regression__ [learners](&Learner.md). 
+This is done by [extracting](functional_data.Rmd#feature-extraction) non-temporal/non-functional features from the curves.
+
+
+## Creating a FDATask
+
+The first step is to get the data in the right format. [%mlr] expects a  [data.frame](&base::data.frame) which consists of the functional features and the target variable as input. The rows are considered independent, while column-ranges, such as $3:136$ in the example below correspond to a functional feature.
+After that a [FDATask](&makeFDATask) that contains the data in a well-defined format is created. [FDATasks](&makeFDATask), just like normal [Tasks](&Task.md) come in different flavours, such as [FDAClassifTask](&makeFDAClassifTask) and [FDARegrTask](&makeFDARegrTask), which can be used according to the class of the target variable.
+
+In the following example, this is done for the [fuelSubset](fuelSubset.task) data from package [%FDboost].
+The data is provided in the following structure:
+
+```{r}
+library(FDboost)
+data(fuelSubset)
+str(fuelSubset)
+```
+
+* __heatan__ is the target variable, in this case a numeric value.
+* __h2o__ is an additional scalar variable.
+* __NIR__ and __UVVIS__ are matricies containing the curve data. Each column corresponds to a single time point the data was sampled at. Each row indicates a single curve. __NIR__ was measured at $231$ time points, while __UVVIS__ was measured at $129$ time points.
+* __nir.lambda__ and __uvvis.lambda__ are vectors of length $231$ and $129$ indicate the time points the data was measured at. Each entry corresponds to one column of __NIR__ and __UVVIS__ respectively.
+
+```{r}
+# Plot NIR curve for first observarion
+library(ggplot2)
+df = data.frame("NIR_obs_1" = fuelSubset$NIR[1, ],
+                "lambda" = fuelSubset$nir.lambda)
+ggplot(df) + 
+  geom_line(aes(y = NIR_obs_1, x = lambda))
+```
+
+The [FDATask](&makeFDATask) can then be constructed as follows:
+
+* __fd.features__ is a list of functional features and the associated column indices.
+* __fd.grids__ is a list of the functional features' time points, i.e when they were measured. 
+
+```{r}
+## Put all values into a data.frame
+df = data.frame(fuelSubset[c("heatan", "h2o", "UVVIS", "NIR")])
+## Change row names to V1 to V367
```

Comment:
So the original structure is a list.
It has one list element for each feature, such as
h2o which is a vector of length NObs
NIR and UVVIS which are matrices of Size NObs x P1, NObs x P2
NIR.lambda and UVVIS.lambda which are vectors of Size P1 and P2. 
-> What  I do is to actually `cbind` those vectors, matricies. from a list of objects that do not have
equal dimensions.
If you have a better idea feel free to tell me :)